index.js

```diff
@@ -1 +1,141 @@
-// implement your API here
+const express = require("express");
+const cors = require("cors");
+const db = require("./data/db");
+
+const server = express();
+
+server.use(cors());
+server.use(express.json());
+
+server.get("/api/users", getAllUsers);
+server.get("/api/users/:id", getUserById);
+server.post("/api/users", postNewUser);
+server.delete("/api/users/:id", deleteUser);
+server.put("/api/users/:id", updateUser);
+server.get("*", handleDefault);
+
+function updateUser(req, res) {
+  const { id } = req.params;
+  const user = req.body;
+
+  if (!id) {
+    res.status(404).json({
+      success: false,
+      message: "The user with the specified ID does not exist."
+    });
+  } else if (!user.name || !user.bio) {
+    res.status(400).json({
+      success: false,
+      message: "Please provide name and bio for the user"
+    });
+  } else {
+    db.update(id, user)
+      .then(updated => {
+        res.status(200).json({
+          success: true,
+          message: "User updated",
+          updated
+        });
+      })
+      .catch(err => {
+        res.status(500).json({
+          success: false,
+          message: "The user information cound not be modified."
+        });
+      });
+  }
+}
+
+function deleteUser(req, res) {
+  const { id } = req.params;
+
+  db.remove(id)
+    .then(deleted => {
+      if (deleted) {
+        res.status(204).end();
+      } else {
+        res.status(404).json({
+          success: false,
+          message: "The user with the specified ID does not exist."
+        });
+      }
+    })
+    .catch(err => {
+      res.status(500).json({
+        success: false,
+        message: "The user could not be removed"
+      });
+    });
+}
+
+function postNewUser(req, res) {
+  const userInfo = {
+    name: req.body.name,
+    bio: req.body.bio
+  };
+
+  if (!userInfo.name || !userInfo.bio) {
+    res.status(404).json({
+      success: false,
+      message: "Please provide name and bio for the user."
+    });
+  }
```

Comment:
I'm still able to add a user without any values in `name` & `bio`, because `db.insert` is still allowed to run even after a response has been sent back to client. To avoid this, just add `else` clause, so it would run only if the if clause fails.